### PR TITLE
#4: ODP: Refactor/odp list and validate original data points 806

### PR DESCRIPTION
**Pull request by sorja**
Part of #806

### CODE REVIEW

File: server/service/odp/listAndValidateOriginalDataPoints.ts [hunk 1]
```diff
@@ -1,3 +1,13 @@
...
+import { BaseProtocol, DB } from '@server/db'
 
-export const { listAndValidateOriginalDataPoints } = OdpRepository
+export const listAndValidateOriginalDataPoints = async (
+  options: { countryIso: CountryIso },
+  client: BaseProtocol = DB
+) => {
+  const { countryIso } = options
+  const odps = await OdpRepository.listOriginalDataPoints({ countryIso }, client)
+  return odps.map((odp: any) => ({ ...odp, validationStatus: validateDataPoint(odp) }))
```
Comment (minotogna) [inline server/service/odp/listAndValidateOriginalDataPoints.ts:12]: no any please,
also return types from all functions

Comment (sorja) [reply server/service/odp/listAndValidateOriginalDataPoints.ts]: Return types will be added in next PR

File: server/service/assessment/ndp/ndpExporter.ts [hunk 2]
```diff
@@ -22,8 +23,8 @@ export const normalizeValue = R.pipe(
   R.join(' ')
 )
 
-export const getCountryData = async (country: any) => {
-  const dataPoints = await OdpService.listOriginalDataPoints(country.countryIso)
+export const getCountryData = async (country: Country) => {
+  const dataPoints = await OdpService.listOriginalDataPoints({ countryIso: country.countryIso })
 
   const result: any = []
```
Comment (minotogna) [inline server/service/assessment/ndp/ndpExporter.ts:29]: no any

File: server/repository/odp/listOriginalDataPoints.ts [hunk 3]
```diff
@@ -1,17 +1,24 @@
...
     SELECT p.id as odp_id
     FROM ${tableName} p
     WHERE country_iso = $1
   `,
     [countryIso]
   )
-  const odps = await Promise.all(res.rows.map((r: any) => getOdp(r.odp_id, schemaName)))
-  return [...odps].sort((odp1, odp2) => odp1.year - odp2.year)
+
+  const odps = await Promise.all(res.map((r: any) => getOdp(r.odp_id, schemaName)))
```
Comment (minotogna) [inline server/repository/odp/listOriginalDataPoints.ts:22]: no any

**Review by minotogna — CHANGES_REQUESTED** [review]

**Review by minotogna — APPROVED** [review]

**Pull request merged**